### ISSUE TITLE
`moderncvcolor` defines colors but doesn't use them

### ISSUE THREAD

**Issue by Baharis** [moderncv/moderncv#215]
**Describe the bug**
After selecting a valid color scheme using `\moderncvcolor{}`, the colors are set but remain unused by the class. Compiled document is black and white. With a help of `xcolor`'s `\extractcolorspecs` one can learn that `color0`, `color1` and `color2` are defined, but none of the elements in the file are colored. Even though `xcolor` knows about  `color0`, `color1` and `color2` from the `moderncvcolor#` file (where # stands for color), defining them explicitly again in `moderncv.cls` DOES turn them on, fixing the issue.

What also fixes the issue (without modifying the source code) is switching the order of commands `moderncvcolor` and `moderncvstyle` in the `tex` file: if the latter precedes the former, all of the color in the document gets lost. The command `moderncvcolor` must precede `moderncvcolor`.

**Expected Behavior**
Declaring `moderncvstyle` before `moderncvcolor` does not affect cause the document to lose all of its color.

**Actual Behavior**
Declaring `moderncvstyle` before `moderncvcolor` does affect cause the document to lose all of its color.

**Minimum Working Example to Reproduce the Problem**
Declare `moderncvstyle` after `moderncvcolor` (i.e. switch the order of lines 11 and 13 in the template around).

**Specifications**
  - moderncv version and source: v2.4.1, CTAN
  - LaTeX compiler (pdfLaTeX, XeLaTeX, etc.) and version: pdfLaTeX from MiKTeK
  - Platform (operating system, LaTeX editor, etc.): Windows, PyCharm

**Additional context**
I have used a modified version of a `moderncv` package for at least 5 years and compiled it mostly on my old computer. I encountered this bug only after installing new MiKTeX on a fresh computer with all the updated packages. Behavior described above is expected, as the current version of `\moderncvstyle` looks for definitions of  `color0`, `color1` and `color2` (which hence must be declared previously in "moderncvcolor") and immediately uses them to declare new color.

However, this didn't use to be the case. The former versions of `moderncv` didn't rely on the order of these commands. What is more offending, the previous template introduced `moderncvstyle` and `moderncvcolor` IN THIS ORDER, meaning that all documents prepared using an old template will become black-and-white when switching to the new version of the package. I am thus writing this up:

- to suggest the template colors be declared not directly in `moderncvcolor`, but later.
- to save time anyone else migrating to a newer version of the package.

Thank you for keeping `moderncv` alive and have a great day!

**Comment by LLdaniel** [comment]
Hi @Baharis, 
thanks for your suggestions! Yes, I see, unfortunately this is a breaking change at the moment introduced with #151 for a more detailed color control. 
So if I understand you correctly you would like to change the code insofar that the order of `\moderncvcolor` and `\moderncvstyle` are irrelevant again. 

I think that would be an option to work on, yes. So I will mark this as an *enhancement*.

**Comment by cutlery** [comment]
Yes, I also would like to see this enhancement (bump up ;-)).  Recently I noticed this regression, and had to put \moderncvcolor{purple} before \moderncvstyle{casual}.
Originally I put it the other way around, as I assumed that the CV style would come with its own default colour, and I then override it with \moderncvcolor.

Cheers.

**Comment by ptoche** [comment]
That wasn't an easy one to find! 

Note to the puzzled:

```
\moderncvcolor{blue}% >2024: \moderncvcolor must go before \moderncvstyle
\moderncvstyle{banking}
```

**Comment by blhiggins** [comment]
Just to add a note that the User Guide still gives the example for these commands in [style-then-color order](https://github.com/moderncv/moderncv/blob/800b295405fabc53b4e95d6ee4cbad63a01501da/manual/moderncv_userguide.tex#L245) in the Basic sectup section, which of course doesn't work. Only if you scroll to the Adjusting colors section later on is there mention of this ordering requirement.

**Comment by thymaro** [comment]
Mentioning the caveat in the documentation on page 3 instead of 10 would do much goodness, already. I imagine the current situation drives away users of moderncv when they can't figure it out, not reaching page 10 of the documentation or not finding the right question that mentions the solution on tex.se

**Comment by PhoenixIV** [comment]
I tried to update an old CV of mine and ran into this issue. I was quite frustrated for 15 minutes before I found the solution here in the issue tracker. Therefore only fixing this in the Docs seems to be a weird solution to me. As this is a breaking change and no one should, by today’s standard, use `moderncvstyle` before `moderncvcolor`, I'd say either make it indifferent again or, if easier, put out an error.

Is there an intended reason to increase usefulness to the user why the order matters now?
If there is a use case for `moderncvstyle` before `moderncvcolor` (multiple documents in one?) put out a warning?

**Comment by LLdaniel** [comment]
Sorry, that you had to find out the hard way about this breaking change. I know, this can be annyoing. But as it is the nature of breaking changes, it breaks things and the only way is to prevent breaking changes or at least document them.

As far as I can see this is done in various places like below and has to be only read:
- RSS feed on [CTAN](https://ctan.org/ctan-ann/id/ajeCtvYWS6BwoVj0@prptp)
- see `CHANGELOG`https://github.com/moderncv/moderncv/blob/d4e882caf5d5fe2f6caf11257b19ae2ba0a7fe64/CHANGELOG#L18
- cf. user guide https://github.com/moderncv/moderncv/blob/d4e882caf5d5fe2f6caf11257b19ae2ba0a7fe64/manual/moderncv_userguide.tex#L731
- pinned ticket

Nevertheless, the reason for the breaking changes was the advantage to have finer control of colors in single CV elements (see #151).

This ticket is still open to resolve the issue and make it indifferent against the order again. I did not have a detailed look into it so far, but contributions are very welcome. If I can help, just let me know.
